### ISSUE TITLE
partitioning and externally stored entity sets

### ISSUE THREAD

**Issue by nlelong** [SCOREC/core#323]
Hello,
I'm new to PUMI, and i'm trying to plug it in our mechanics code to benefit from its distributed capabilities. We're currently using an other mesh library which has some limitations and wish to test PUMI as replacement.
The majority of our meshes are in a specific mesh format. Our Mesh class encapsulates an apf::Mesh2 and Vertices and elements are added from reading coordinates and connectivities. We also define regions to apply some intial/boundary conditions and for some post-treatment. These regions are read from the mesh and are a list of entities (basically it's a std::vector of pMeshEnt). 

Without partitioning, it seems to work fine, I get good adjacencies and manage to run my computations. Then I tried to add some partitoning, following the split.cc test case as example. 
I then had errors when I browsed my regions when I asked for adjacencies for the saved mesh entities. I found out that, while the partitioning worked fine, the pMeshEnt values saved didn't match the same thing anymore. 

My question is then : how can I keep the regions information during partitioning ? Should I work with GlobalID ? Should I use tags ?
I'm aware I'm not using PUMI in a nominal way, because I'm first trying to plug PUMI into an existing more general interface. I don't use at all the geometric model interface.

Thanks a lot for your help and advice.

Nicolas

**Comment by cwsmith** [comment]
Hi Nicolas,

The use of these regions is not quite clear to me.  It sounds like they should be geometric model entities given that they have boundary/initial conditions applied to them.  Mesh classification on the model (i.e, the association of mesh entities to geometric model entities) could then be queried to determine which boundary/initial conditions need to be applied and to handle any post processing.  

An alternative to creating the model entities and classifying the mesh against them would be mesh entity tags (as you suggested). The tags are migrated with the mesh entities during partitioning and are synchronized across partiition boundaries (i.e., all copies of a given mesh entity will have the same tag value).

-Cameron

**Comment by nlelong** [comment]
Thanks Cameron for you answer.
Yes I think that in the PUMI framework that should correspond to geometric model entities. However, I don't quite understand how to create them from an existing mesh. How are generated the geometric model files you use ? 
For example, when we read from Gmsh mesh files, we read the physical entities in gmsh to define our regions (boundaries, etc...). I saw that in the pumi gmsh reader, the tags are discarded and the geometric model comes from a model file. Where does it come from ? It's probable I'm missing something here...

I tried to tag my regions before partitioning and if I'm not through it, it seems promising.
